python/hail/dataset.py

```diff
@@ -915,6 +915,278 @@ def annotate_variants_vds(self, other, expr=None, root=None):
+
+            vds = vds.annotate_variants_db(['va.gene.constraint.pli'], gene_key = 'va.gene_symbol')
+
+        **Notes**
+
+        Annotations in the database are bi-allelic, so splitting multi-allelic variants in the VDS before using this 
+        method is recommended to capture all appropriate annotations from the database. To do this, run :py:meth:`split_multi` 
+        prior to annotating variants with this method:
+
+        .. code-block:: python
+
+            vds = vds.split_multi().annotate_variants_db(['va.cadd.RawScore', 'va.cadd.PHRED'])
+
+        :param annotations: List of annotations to import from the database.
+        :type annotations: str or list of str 
+
+        :param gene_key: Existing variant annotation used to map variants to gene symbols if importing gene-level 
+            annotations. If not provided, the method will add VEP annotations and parse them as described in the 
+            database documentation to obtain one gene symbol per variant.
+        :type gene_key: str
+
+        :return: Annotated variant dataset.
+        :rtype: :py:class:`.VariantDataset`
+        """
+
+        # import modules needed by this function
+        import json
+        import urllib2
+        
+        # print warning if analysis VDS is not split
+        if not self.was_split():
+            print('Warning: VDS not split. Multi-allelic sites will not be annotated.')
+
+        # ensure annotations object is a list
+        if type(annotations) == str:
+            annotations = [annotations]
+
+        # collect annotations in a set
+        annotations = set(annotations)
+
+        # read in file map from bucket
+        files = json.loads(urllib2.urlopen('https://storage.googleapis.com/annotationdb/file_map.json?ignoreCache=1').read())
+
+        # build dictionary of file: expr for each database file that needs to be read, based on the annotations provided
+        file_exprs = {}
+        for f in files:
+
+            # dictionary of annotation: expr for each underlying file in the database
+            # e.g: file_annotations is {'va.cadd': 'vds', 'va.cadd.RawScore': 'vds.RawScore', 'va.cadd.PHRED': 'vds.PHRED'}
+            file_annotations = f['annotations']
+
+            # "depths" of each annotation
+            # e.g: {'va.cadd': 2, 'va.cadd.RawScore': 3, 'va.cadd.PHRED': 3}
+            depths = {k: len(k.split('.')) for k, _ in file_annotations.iteritems()}
+
+            # maximum "depth"
+            # e.g: 3
+            max_depth = max(depths.values())
+
+            # subset to annotations explicitly in the database file
+            # e.g: [('va.cadd.RawScore', 'vds.RawScore'), ('va.cadd.PHRED', 'vds.PHRED')]
+            terms = [x for x in file_annotations.items() if depths[x[0]] == max_depth]
+
+            # get the "root" annotation of the underlying database file
+            # e.g: 'va.cadd'
+            root = [x for x in file_annotations.iteritems() if depths[x[0]] == max_depth - 1][0][0]
```

Comment:
Yes, you're right. I think can even just iterate through `file_annotations` and it should grab the keys by default.